OpenRailAssociation/osrd
# Add ERTMS-ETCS brake params and signaling to rolling-stock model

Note: not managing towed rolling-tock so far (may not have a max_speed, not sure how to merge brake values).

Also:
* Add ETCS_LEVEL2 signaling for infra and rolling-stock (complete duplicate of TVM430 for now).
* Compute ETCS_LEVEL2 ranges of the path
* Fix some small bits (🔍 review by commit)
* Tests:
  * add tests ETCS_LEVEL2 rolling-stock (derived from fast_rolling_stock).
  * add tests ETCS infra (including electrical_profile). Derived from small_infra, just switching all the signaling from BAL to
ETCS_LEVEL2.

✅ Hand-tested only for now:
* keeps ETCS params when editing RS
* when removing the `TODO()` on core ETCS-brake part processing, it "works" as expected, with the `TODO()` it fails in core

fix #9708 and #9706

## TODO after this PR:
* add integration tests (circumvent ETCS intervals for now with a TODO) > For this PI : https://github.com/OpenRailAssociation/osrd/pull/10294
* prioritize signaling systems for blocks of pathfinding: ETCS>TVM430>TVM300>BAL>BAPR
* use the signaling system output from pathfinding everywhere (remove recoverBlockPath - only deprecated now)
* merge small_infra BAL and ERTMS (and TVM?)
